src/app/icd/client/DefaultCheckInDelegate.cpp

```diff
@@ -0,0 +1,50 @@
+/*
+ *    Copyright (c) 2023 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include "CheckInHandler.h"
+#include <app/icd/client/DefaultCheckInDelegate.h>
+#include <crypto/CHIPCryptoPAL.h>
+#include <lib/support/CodeUtils.h>
+#include <lib/support/logging/CHIPLogging.h>
+
+namespace chip {
+namespace app {
+
+CHIP_ERROR DefaultCheckInDelegate::Init(ICDClientStorage * storage)
+{
+    VerifyOrReturnError(storage != nullptr, CHIP_ERROR_INVALID_ARGUMENT);
+    VerifyOrReturnError(mpStorage == nullptr, CHIP_ERROR_INCORRECT_STATE);
+    mpStorage = storage;
+    return CHIP_NO_ERROR;
+}
+
+void DefaultCheckInDelegate::OnCheckInComplete(ICDClientInfo & clientInfo)
+{
+    ChipLogProgress(
+        ICD, "Check In Message processing complete: start_counter=%" PRIu32 " offset=%" PRIu32 " nodeid=" ChipLogFormatScopedNodeId,
+        clientInfo.start_icd_counter, clientInfo.offset, ChipLogValueScopedNodeId(clientInfo.peer_node));
+}
+
+void DefaultCheckInDelegate::OnRefreshKey(ByteSpan & keyData)
+{
+    uint8_t randomGeneratedSymmetricKey[chip::Crypto::kAES_CCM128_Key_Length];
+    chip::Crypto::DRBG_get_bytes(randomGeneratedSymmetricKey, sizeof(randomGeneratedSymmetricKey));
+    keyData = ByteSpan(randomGeneratedSymmetricKey);
```

Comment:
This right here...  Please fix this code.  @yunhanw-google @mkardous-silabs, I really hope this code wasn't here when you reviewed....

@thivya-amazon please figure out why this code is not ok and don't ever write this sort of thing again.

Reply:
@bzbarsky-apple Totally my fault for assuming assignment operator in ByteSpan would do a deep copy.I checked the definition in Span.h and it only does a shallow copy. So, keyData will have garbage value ater exiting OnRefreshKey.

Reply:
Right. It's just like `std::span`.  It has no storage associated with it.

keyData will have worse than a garbage value: it will be pointing at the stack, and will allow reading various data from the stack.  And then we will put that data on the wire.